pkg/dns/service.go

```diff
@@ -0,0 +1,272 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	v1 "github.com/Kuadrant/multi-cluster-traffic-controller/pkg/apis/v1"
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/dns/aws"
+	"github.com/Kuadrant/multi-cluster-traffic-controller/pkg/traffic"
+	"github.com/lithammer/shortuuid/v4"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	labelRecordID = "kuadrant.io/record-id"
+)
+
+var AlreadyAssignedErr = fmt.Errorf("managed host already assigned")
+
+type Service struct {
+	controlClient client.Client
+	// this is temporary setting the tenant ns in the control plane.
+	// will be removed when we have auth that can map to a given ctrl plane ns
+	defaultCtrlNS string
+
+	hostResolver HostResolver
+}
+
+func NewService(controlClient client.Client, hostResolv HostResolver, defaultCtrlNS string) *Service {
+	return &Service{controlClient: controlClient, defaultCtrlNS: defaultCtrlNS, hostResolver: hostResolv}
+}
+
+func (s *Service) resolveIPS(ctx context.Context, t traffic.Interface) ([]string, error) {
+	activeDNSTargetIPs := []string{}
+	targets, err := t.GetDNSTargets()
+	if err != nil {
+		return nil, err
+	}
+	for _, target := range targets {
+		if target.TargetType == v1.TargetTypeIP {
+			activeDNSTargetIPs = append(activeDNSTargetIPs, target.Value)
+			continue
+		}
+		addr, err := s.hostResolver.LookupIPAddr(ctx, target.Value)
+		if err != nil {
+			return activeDNSTargetIPs, fmt.Errorf("DNSLookup failed for host %s : %s", target.Value, err)
+		}
+		for _, add := range addr {
+			activeDNSTargetIPs = append(activeDNSTargetIPs, add.IP.String())
+		}
+	}
+	return activeDNSTargetIPs, nil
+}
+
+func (s *Service) GetManagedRecords(ctx context.Context, traffic traffic.Interface) ([]*v1.DNSRecord, error) {
```

Comment:
Why `GetManagedRecords` and not `GetDNSRecords` since that's what it's returning?

Reply:
Fair. Will change